### DISCUSSION TITLE
How do I fix this?

### DISCUSSION THREAD

**Discussion by Amaranth-101** [community/community#138191] [Programming Help]
### How do I fix this?

<img width="1440" alt="Screenshot 2024-09-09 at 10 06 29 AM" src="https://github.com/user-attachments/assets/9ad7584f-0a35-4375-ae43-a5308d65628d">

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Y3llow45** [comment]
You haven't defined report_to_file(). Just make sure you have def report_to_file(self): #code...

**Comment by lucaskrol990** [comment]
Can you show your definition of report_to_file()?

**Reply by Y3llow45** [reply]
Your function should look something like this:
```
def report_to_file(self):
    self.filename = f"log_{self.start_dt.strftime('%Y-%m-%d_%H-%M-%S')}.txt"
    # Open the file in append mode, create it if doesn't exist
    with open(self.filename, "a") as file:
        file.write(self.log)
    self.log = ""
```
